packages/e2e-test-utils-playwright/src/page/insert-block.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @typedef {Object} BlockRepresentation
+ * @property {string}                 name        Block name.
+ * @property {?Object}                attributes  Block attributes.
+ * @property {?BlockRepresentation[]} innerBlocks Nested blocks.
+ */
+
+/**
+ * @this {import('./').PageUtils}
+ * @param {BlockRepresentation} blockRepresentation Inserted block representation.
+ */
+async function insertBlock( blockRepresentation ) {
```

Comment:
Is this meant to be the playwright version of this one: https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-test-utils/src/inserter.js#L165? 

If yes, we're bypassing all the insertion flow from the inserter and doesn't seem something e2e tests should be doing. The good thing about the original util is that it's using the normal user flow with the inserter and we can catch and regressions.

Reply:
I get what you mean, but still I don't think this is the same with using requestUtils to create/delete posts etc.. and the main example from Kent is logging a user. This flow though handles other things useful for many tests, like where a block should be inserted(rootClientId), focus, selection etc.. which they may be needed in many tests. An example would be to insert blocks -> navigate between and insert other blocks.

My personal opinion is that we could have a util like this with a different name and since we're doing a migration to playwright we should have one similar to the previous one(that follows the flow). This way a user could have the option to select what they want. Also if we care about speed and want to just have some content, we could add this content in the request when we create a post, no?

Reply:
> I don't think this is the same with using requestUtils to create/delete posts etc.. and the main example from Kent is logging a user.

I don't think these are that different though. Neither of them is the main thing being tested in their test cases. Just that inserting blocks is faster than logging in a user.

> This flow though handles other things useful for many tests, like where a block should be inserted(rootClientId), focus, selection etc.. which they may be needed in many tests

Do you have an example where it isn't covered in the dedicated `inserting-blocks` test suite? If this is something that depends on the inserted blocks then I think it's worth it to add it to the utils as you said. But then I think there might be a bigger issue in our architecture where the flow would change if inserting a different block.

> we could add this content in the request when we create a post, no?

True! FWIW, there's a new util added recently called `editor.setContent` that can set and replace the full content in the editor. I think it should be used when we're just setting/resetting the editor, but not all of the tests have been migrated yet.

> since we're doing a migration to playwright we should have one similar to the previous one(that follows the flow).

IMHO, not every util should have its Playwright version of the same thing. I'd prefer [inlining utils](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#dont-overload-test-utils-inline-simple-utils) or using POM to encapsulate them in their belonging test suite. This encourages best practices and discourages repeating ourselves by accidentally using the wrong utils. We're also trying to refactor the tests during the migration to hopefully leverage the full power of Playwright.

Full coverage isn't a priority of these e2e tests, and I'd favor explicit assertions over implicit flow hidden in e2e utils. If it's something special enough to be tested, then I think we should write a test case for it and do the normal flow explicitly.

This, however, is just my own preference. I don't know what everyone else thinks of this though. I sort of just took the liberty to write the best practices guide because there wasn't anyone else opposing it either. Happy to discuss this further if you feel strongly about this :)

Reply:
>Do you have an example where it isn't covered in the dedicated inserting-blocks test suite? If this is something that depends on the inserted blocks then I think it's worth it to add it to the utils as you said. But then I think there might be a bigger issue in our architecture where the flow would change if inserting a different block.

I don't have something specific in mind, but I'd guess there are some tests with my example: `An example would be to insert blocks -> navigate between and insert other blocks.` Maybe there are in the remaining tests and this will be visible when we migrate those..

I don't have a strong opinion right now. The consistency between the two is what I'd expected at first, but this isn't necessarily something we aim to, as they are different packages.

Reply:
> An example would be to insert blocks -> navigate between and insert other blocks

Ahh, I think I know what you mean here. I'm not sure though, I don't think that's mostly relevant to the block we're testing. It would be nice to provide an option to call `insertBlock` at any given position though.

> The consistency between the two is what I'd expected at first, but this isn't necessarily something we aim to, as they are different packages.

Yep, the `e2e-test-utils-playwright` package is not meant to be as a drop-in replacement for `e2e-test-utils`. It's a package to help us write e2e tests with Playwright. I understand that it's a common misconception though since the term "migration" implies a lot.